libcaf_net/src/net/web_socket/lower_layer.cpp

```diff
@@ -20,7 +20,9 @@ void lower_layer::shutdown() {
 }
 
 void lower_layer::shutdown(const error& reason) {
-  if (reason.code() == static_cast<uint8_t>(sec::protocol_error)) {
+  if (reason.code() == static_cast<uint8_t>(sec::connection_closed)) {
+    shutdown(status::normal_close, to_string(reason));
```

Comment:
Can this mapping stay: If I (the API user) shut down the connection with `connection_closed`, I mean a normal close and stop responding, with some message. 
I could keep this mapping only in the `framing` layer, and remove it from the public API . I saw that calling only `shutdown()` without arguments closes with `normal_close`.

Reply:
I'm not sure what your question is. You mean removing this function? It's inherited from `generic_lower_layer`, so we can't drop it if that's your question.

Reply:
The question really was weather this is OK: the user calls `shutdown` with the error code  `sec::connection_closed` and the shutdown function maps ti to `web_socket::status::normal_close`.

Reply:
Yes. The WebSocket layer should map "generic" CAF codes to the protocol-specific codes as well as it reasonably can.